### PR TITLE
GridLayout refinements + fixes

### PR DESCRIPTION
**Pull request by jessesquires**
- Clean up GridLayout + docs
- Remove GridLayout scroll direction, not used/supported?
- Fix some file target membership issues
- Add GridLayout to umbrella header
- "Sort by name" on xcode project directories
- crtl-I auto format in xcode
- re-gen docs

### CODE REVIEW

File: Source/IGListGridCollectionViewLayout.h [hunk 1]
```diff
@@ -9,18 +9,15 @@
...
 
 /**
- The scroll direction of the grid.
+ `IGListGridCollectionViewLayout` provides a vertically-scrolling, section-based grid layout for `UICollectionView`.
+ Items in the layout are displayed consecutively in a grid with exactly 1 item per section.
+ If items are square, the appearance would be similar to the iOS Photos app.
+ However, the size of the items for each section can vary.
  */
-@property (nonatomic, assign) UICollectionViewScrollDirection scrollDirection;
+@interface IGListGridCollectionViewLayout : UICollectionViewLayout
```
Comment (rnystrom) [inline Source/IGListGridCollectionViewLayout.h]: Wanna toss `IGLK_SUBCLASSING_RESTRICTED` on here too? Simple layout, def not meant to be subclassed.

File: Source/IGListGridCollectionViewLayout.m [hunk 2]
```diff
@@ -61,11 +64,10 @@ - (instancetype)initWithCoder:(nonnull NSCoder *)aDecoder {
 }
 
 - (void)commonInit {
-    _scrollDirection = UICollectionViewScrollDirectionVertical;
     _minimumLineSpacing = 0.0;
     _minimumInteritemSpacing = 0.0;
-    _lineCache = [NSMutableArray<_IGListGridLayoutLine *> array];
-    _lineForItem = [NSMutableArray array];
+    _lineCache = [NSMutableArray new];
+    _lineForItem = [NSMutableArray new];
```
Comment (rnystrom) [inline Source/IGListGridCollectionViewLayout.m]: 👌

File: Source/IGListKit.h [hunk 3]
```diff
@@ -28,6 +28,7 @@ FOUNDATION_EXPORT const unsigned char IGListKitVersionString[];
 #import <IGListKit/IGListDiffable.h>
 #import <IGListKit/IGListDisplayDelegate.h>
 #import <IGListKit/IGListExperiments.h>
+#import <IGListKit/IGListGridCollectionViewLayout.h>
```
Comment (rnystrom) [inline Source/IGListKit.h]: Ahhh yes

**Comment by jessesquires** [comment]
Follow up from #225

@zhubofei -- Can you confirm the behavior of the GridLayout? It looks like scroll direction was not actually supported yet. Is this correct?

This is what the docs stated, and it looked like it wasn't used. Ripped it out for now. (Can't ship incomplete APIs.) We can always add this later.

**Comment by zhubofei** [comment]
@jessesquires Yes, scroll direction is not supported for now. I plan to work on that this week.

**Comment by jessesquires** [comment]
@zhubofei 👍 Sounds good!

We'll probably leave any major changes (like that) for post-2.0 😄

**Comment by jessesquires** [comment]
replaced by #246

**Pull request closed without merging**